pkg/ccl/multitenantccl/tenantcostclient/tenant_side.go

```diff
@@ -27,94 +28,55 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/metric"
 	"github.com/cockroachdb/cockroach/pkg/util/stop"
-	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
 	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/errors/errorspb"
 )
 
+// ProvisionedVCPUs limits the amount of CPU that can be used by a tenant. If
+// zero, then provisioned vCPUs are not being used by the tenant (e.g. it's an
+// on-demand tenant).
+var ProvisionedVCPUs = settings.RegisterIntSetting(
```

Comment:
I expected this to be a bool setting instead of rate setting. How will `tenant_cost_control.provisioned_vcpus` and the rate defined in the distributed token bucket interact?

Reply:
We need the # provisioned vCPUs in order to compute the model. In particular, the write QPS is per-node, and to determine the number of nodes, we need to divide the number of provisioned vCPUs by `vCPUsPerNode`. The `ProvisionedVCPUs` setting is expected to be overridden in the system tenant so that it cannot be updated by the tenant. I was expecting an outside caller (i.e. Intrusion) to keep this setting in sync with the distributed token bucket. I suppose we could also consider automatically setting the distributed token bucket when `ProvisionedVCPUs` is set. What do you think?